src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -344,6 +344,44 @@ impl<'a, T: Timestamp + Lattice> StorageControllerMut<'a, T> {
                 );
         }
     }
+    async fn recv(&mut self) -> Result<Option<StorageResponse<Self::Timestamp>>, anyhow::Error> {
+        self.state.client.recv().await
+    }
+}
+
+// Internal interface
+impl<T: Timestamp + Lattice> Controller<T> {
+    /// Create a new storage controller from a client it should wrap.
+    pub fn new(client: Box<dyn StorageClient<T>>) -> Self {
```

Comment:
is this internal if its `pub`? The comment above the impl block might be inaccurate

Reply:
It's an inaccurate comment, good catch!